ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -583,6 +583,17 @@ func (r *RayClusterReconciler) reconcileServeService(ctx context.Context, instan
 }
 
 func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv1.RayCluster) error {
+	// if RayCluster is suspended, delete all pods and skip reconcile
+	if instance.Spec.Suspend != nil && *instance.Spec.Suspend {
+		// State is now suspended, which means all pods have been deleted already
+		if instance.Status.State == rayv1.Suspended {
+			return nil
+		}
+
+		clusterLabel := client.MatchingLabels{common.RayClusterLabelKey: instance.Name}
```

Comment:
@kevin85421 is it safe to use this cluster label to find pods to terminate for suspension? Are there pods we shouldn't delete using this label if a RayCluster is suspended?

Alternatively we can specifically call DeleteCollection for head node and worker group nodes like this: 
```
headPodLabels := client.MatchingLabels{common.RayClusterLabelKey: instance.Name, common.RayNodeTypeLabelKey: string(rayv1.HeadNode)}
if err := r.DeleteAllOf(ctx, &corev1.Pod{}, client.InNamespace(instance.Namespace), headPodLabels); err != nil {
    return err
}

for _, worker := range instance.Spec.WorkerGroupSpecs {
	workerGroupLabels := client.MatchingLabels{common.RayClusterLabelKey: instance.Name, common.RayNodeGroupLabelKey: worker.GroupName}
	if err := r.DeleteAllOf(ctx, &corev1.Pod{}, client.InNamespace(instance.Namespace), workerGroupLabels); err != nil {
		return err
	}
}
```

Reply:
> is it safe to use this cluster label to find pods to terminate for suspension? Are there pods we shouldn't delete using this label if a RayCluster is suspended?

Yes, I think it is safe. 

> Alternatively we can specifically call DeleteCollection for head node and worker group nodes like this

SGTM